flaps/flaps_machines.go

```diff
@@ -17,64 +17,9 @@ import (
 
 var NonceHeader = "fly-machine-lease-nonce"
 
-type flapsActionInfo struct {
-	action          flapsAction
-	machineID       string
-	queryParameters url.Values
-	headers         map[string][]string
-}
-
-func flapsActionInfoToEndpoint(actionInfo flapsActionInfo, appName string) (string, error) {
-	var endpoint string
-
-	if callEndpoint, ok := flapsActionToEndpoint[actionInfo.action]; ok {
-		if actionInfo.machineID != "" {
-			endpoint = fmt.Sprintf("/%s", actionInfo.machineID)
-		}
-		if callEndpoint != "" {
-			endpoint += fmt.Sprintf("/%s", callEndpoint)
-		}
-
-		queryParams := actionInfo.queryParameters.Encode()
-
-		endpoint = fmt.Sprintf("%s?%s", endpoint, queryParams)
-		endpoint = fmt.Sprintf("/apps/%s/machines%s", appName, endpoint)
-
-	} else {
-		return "", fmt.Errorf("flaps action %s (%d) does not exist", &actionInfo.action, actionInfo.action)
-	}
-
-	return endpoint, nil
-}
-
-func (f *Client) sendRequestMachines(ctx context.Context, actionInfo flapsActionInfo, in, out interface{}) error {
-	endpoint, err := flapsActionInfoToEndpoint(actionInfo, f.appName)
-	if err != nil {
-		return err
-	}
-
-	var method string
-	var ok bool
-	if method, ok = flapsActionToMethod[actionInfo.action]; !ok {
-		return fmt.Errorf("method not found for action %s (%d)", actionInfo.action.String(), actionInfo.action)
-	}
-
-	startTimer := time.Now()
-	statusCode, err := f._sendRequest(ctx, method, endpoint, in, out, actionInfo.headers)
-	time := time.Since(startTimer)
-
-	defer func() {
-		// If the status code is 0, then we bumped into an error that doesn't involve sending data
-		if statusCode != 0 {
-			sendFlapsCallMetric(ctx, actionInfo.action, statusCode, time)
-		}
-	}()
-
-	if err != nil {
-		return err
-	}
-
-	return err
+func (f *Client) sendRequestMachines(ctx context.Context, action flapsAction, method, endpoint string, in, out interface{}, headers map[string][]string) error {
```

Comment:
The number of args is getting hard to track. Maybe you could cut down on one by doing what the stdlib http client does. Move the http method to the function name, like `sendMachinesRequestPost`, `sendMachinesRequestGet`, and those methods in turn pass the http method into `_sendRequest`.